### PR TITLE
[DOCS] Consolidated upgrade info

### PR DESCRIPTION
**Pull request by debadair**
8.0 upgrade: https://stack-docs_1970.docs-preview.app.elstc.co/guide/en/elastic-stack/8.0/upgrading-elastic-stack.html

Cloud upgrade: https://stack-docs_1970.docs-preview.app.elstc.co/guide/en/elastic-stack/8.0/upgrade-elastic-stack-for-elastic-cloud.html

On-prem upgrade: https://stack-docs_1970.docs-preview.app.elstc.co/guide/en/elastic-stack/8.0/upgrading-elastic-stack-on-prem.html

Elasticsearch upgrade: https://stack-docs_1970.docs-preview.app.elstc.co/guide/en/elastic-stack/8.0/upgrading-elasticsearch.html

Kibana upgrade: https://stack-docs_1970.docs-preview.app.elstc.co/guide/en/elastic-stack/8.0/upgrading-kibana.html

### CODE REVIEW

**Comment by LeeDr** [comment]
The preview version of https://stack-docs_1970.docs-preview.app.elstc.co/guide/en/elastic-stack/8.0/upgrading-elastic-stack.html still shows 8.0.0-rc1.
Other than that, we should add statements about CCS and CCR.  Something along the lines of;

* The 8.0 stack only allows Cross Cluster Search (CCS) to version 7.17 or greater.  See Stack Management > Remote Clusters to see if you are using CCS. See https://www.elastic.co/guide/en/elasticsearch/reference/8.0/modules-cross-cluster-search.html for more details.
* If you are using Cross Cluster Replication (CCR) the cluster containing follower indices must be running the same or newer version of Elasticsearch as the remote cluster.  See https://www.elastic.co/guide/en/elasticsearch/reference/8.0/xpack-ccr.html for version compatibility matrix.

**Comment by LeeDr** [comment]
The preview of https://stack-docs_1970.docs-preview.app.elstc.co/guide/en/elastic-stack/8.0/upgrade-elastic-stack-for-elastic-cloud.html still shows 8.0.0-rc1.

In this page, I would suggest in the `Important` section where it talks about deprecations, maybe add the notes about CCS and CCR compatibility.

File: docs/en/install-upgrade/upgrading-elasticsearch.asciidoc [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+
+
+.. Set `path.logs` in `config/elasticsearch.yml` to point to the location
+   where you want to store your logs. If you do not specify this setting,
+   logs are stored in the directory you extracted the archive to.
+
+[TIP]
+================================================
+
+When you extract the zip or tarball packages, the `elasticsearch-n.n.n`
```
Comment (gchaps) [inline docs/en/install-upgrade/upgrading-elasticsearch.asciidoc:164]: elasticsearch-n.n.n or can you be a little more specific here with the version

**Review by gchaps — APPROVED** [review]
Looks good.  Comments mainly have to do with formatting and links.

**Comment by debadair** [comment]
The RC1 references are due to the attribute used for the version. The attribute is updated as part of the release process.

File: docs/en/install-upgrade/index.asciidoc [hunk 2]
```diff
@@ -22,6 +22,18 @@ include::upgrading-stack-cloud.asciidoc[]
 
 include::upgrading-stack-on-prem.asciidoc[]
 
+include::upgrading-elasticsearch.asciidoc[]
+
+include::upgrading-kibana.asciidoc[]
+
+// include::upgrading-logstash.asciidoc[]
+
```
Comment (karenzone) [inline docs/en/install-upgrade/index.asciidoc:30]: FYI:  WIP for linking is here: https://github.com/elastic/logstash/pull/13705

File: docs/en/install-upgrade/upgrading-elasticsearch.asciidoc [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+even if you opt to do a full-cluster restart instead of a rolling upgrade. 
+This enables you to use the Upgrade Assistant to <<prepare-to-upgrade, prepare to upgrade>>.
+
+An {es} cluster can be upgraded one node at
+a time so upgrading does not interrupt service. Running multiple versions of
+{es} in the same cluster beyond the duration of an upgrade is
+not supported, as shards cannot be replicated from upgraded nodes to nodes
+running the older version.
+
+IMPORTANT: Upgrading from a pre-release to 8.0 GA is not supported.
```
[the remark this answers is not in the document]
Comment (debadair) [reply docs/en/install-upgrade/upgrading-elasticsearch.asciidoc:28]: There are generally only prereleases for major versions, so going to leave this as-is.

File: docs/en/install-upgrade/upgrading-elasticsearch.asciidoc [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+--------------------------------------------------
+POST _ml/set_upgrade_mode?enabled=true
+--------------------------------------------------
++
+When you disable upgrade mode, the jobs resume using the last model
+state that was automatically saved. This option avoids the overhead of managing
+active jobs during the upgrade and is faster than explicitly stopping {dfeeds}
+and closing jobs.
+
+* {ml-docs}/stopping-ml.html[Stop all {dfeeds} and close all jobs]. This option
```
[the remark this answers is not in the document]
Comment (debadair) [reply docs/en/install-upgrade/upgrading-elasticsearch.asciidoc:110]: Good catch!

File: docs/en/install-upgrade/upgrading-stack-cloud.asciidoc [hunk 5]
```diff
@@ -1,77 +1,66 @@
...
-There are two settings that prompt you to update:
+Security realm settings::
+During the upgrade process, you are prompted to update the security realm settings if your `user_settings` include a `xpack.security.authc.realms` value. 
 +
-* If the security realm settings are located in `user_settings_override`, contact support to help you upgrade.
-+
-* If the security realm settings are located in `user_settings`, you'll receive a message that allows you to modify these settings. 
-+
-To modify the `user_settings`:
+If the security realms are configured in `user_settings`, you'll be prompted to modify the settings:
```
[the remark this answers is not in the document]
Comment (debadair) [reply docs/en/install-upgrade/upgrading-stack-cloud.asciidoc:54]: The distinction is *where* the user settings are configured. I changed the first reference to plain text.

File: docs/en/install-upgrade/upgrading-elasticsearch.asciidoc [hunk 6]
```diff
@@ -0,0 +1,361 @@
...
+An {es} cluster can be upgraded one node at
+a time so upgrading does not interrupt service. Running multiple versions of
+{es} in the same cluster beyond the duration of an upgrade is
+not supported, as shards cannot be replicated from upgraded nodes to nodes
+running the older version.
+
+IMPORTANT: Upgrading from a pre-release to 8.0 GA is not supported.
+Pre-releases should only be used for testing in a temporary environment.
+
+When performing a <<rolling-upgrade, rolling upgrade>>:
```
Comment (debadair) [inline docs/en/install-upgrade/upgrading-elasticsearch.asciidoc:31]:
```suggestion
When performing a <<rolling-upgrades, rolling upgrade>>:
```

File: docs/en/install-upgrade/upgrading-kibana.asciidoc [hunk 7]
```diff
@@ -0,0 +1,50 @@
+[[upgrading-kibana]]
+== Upgrade {kib}
+
+To upgrade from 7.16 or earlier to {version}, **you must first upgrade to {prev-major-last}**.
+This enables you to use the Upgrade Assistant to <<prepare-to-upgrade, prepare to upgrade>>.
```
Comment (debadair) [inline docs/en/install-upgrade/upgrading-kibana.asciidoc:5]:
```suggestion
This enables you to use the **Upgrade Assistant** to <<prepare-to-upgrade, prepare to upgrade>>.
```

File: docs/en/install-upgrade/upgrading-kibana.asciidoc [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+{ref}/snapshot-restore.html[backup snapshot] that includes the `kibana` feature
+state. Snapshots include this feature state by default.
+
+For more information, see {kibana-ref}/upgrade-migrations.html[upgrade migrations].
+====
+
+To upgrade {kib}:
+
+. Shut down all {kib} instances. {kib} does not support rolling upgrades.
+Upgrading while older {kib} instances are running can cause data loss or upgrade failures.
```
Comment (debadair) [inline docs/en/install-upgrade/upgrading-kibana.asciidoc:20]:
```suggestion
**Upgrading while older {kib} instances are running can cause data loss or upgrade failures.**
```

File: docs/en/install-upgrade/upgrading-kibana.asciidoc [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+
+.. **Extract the archive to a new directory** to be sure that you
+  don't overwrite the `config` or `data` directories. 
+
+.. Copy the files from the `config` directory from your old installation to your
+  new installation. 
+.. Copy the files from the `data` directory from your old installation to your
+  new installation.
+  +
+  IMPORTANT: If you use {monitor-features}, you must re-use the data directory when you upgrade {kib}.
```
Comment (debadair) [inline docs/en/install-upgrade/upgrading-kibana.asciidoc:40]:
```suggestion
IMPORTANT: If you use {monitor-features}, you must re-use the data directory when you upgrade {kib}. 
```

File: docs/en/install-upgrade/upgrading-stack-cloud.asciidoc [hunk 10]
```diff
@@ -1,77 +1,66 @@
...
 +
 . On the *Update security realm settings* window, edit the settings.
 +
 . Click *Update settings*.
+_
```
Comment (debadair) [inline docs/en/install-upgrade/upgrading-stack-cloud.asciidoc:59]:
```suggestion
```

File: docs/en/install-upgrade/upgrading-stack.asciidoc [hunk 11]
```diff
@@ -1,21 +1,67 @@
 [[upgrading-elastic-stack]]
-== Upgrading the Elastic Stack
+== Upgrade to Elastic {version}
 
-When upgrading to a new version of {es}, you need to upgrade
-each of the products in your Elastic Stack. Beats and Logstash 7.n are
-compatible with {es} {version} to give you flexibility in scheduling
-the upgrade.
+To upgrade to {version} from 7.16 or earlier, **you must first upgrade to {prev-major-last}**.
+This enables you to use the Upgrade Assistant to identify and resolve issues,
```
Comment (debadair) [inline docs/en/install-upgrade/upgrading-stack.asciidoc:5]:
```suggestion
This enables you to use the **Upgrade Assistant** to identify and resolve issues,
```

File: docs/en/install-upgrade/upgrading-stack.asciidoc [hunk 12]
```diff
@@ -1,21 +1,67 @@
...
+You can view your remote clusters from **Stack Management > Remote Clusters**. 
+====
+
+[discrete]
+[[prepare-to-upgrade]]
+=== Prepare to upgrade
+
+. Use the {kibana-ref}/upgrade-assistant.html[Upgrade Assistant] 
+to prepare for your upgrade from {prev-major-last} to {version}.
+The upgrade assistant identifies deprecated settings and guides
```
Comment (debadair) [inline docs/en/install-upgrade/upgrading-stack.asciidoc:34]:
```suggestion
The **Upgrade Assistant** identifies deprecated settings and guides
```

File: docs/en/install-upgrade/upgrading-stack.asciidoc [hunk 13]
```diff
@@ -1,21 +1,67 @@
...
+you through resolving issues and reindexing indices created before 7.0.
+Make sure you have a current snapshot before making configuration changes
+or reindexing.
++
+**You must resolve all critical issues before proceeding with the upgrade.**
+
+. Review the deprecation logs from the Upgrade Assistant to
```
Comment (debadair) [inline docs/en/install-upgrade/upgrading-stack.asciidoc:41]:
```suggestion
. Review the deprecation logs from the **Upgrade Assistant** to 
```

File: docs/en/install-upgrade/upgrading-kibana.asciidoc [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+  Otherwise, the {kib} instance is assigned a new persistent UUID and becomes a new instance in the monitoring data.
+
+.. Install the appropriate versions of all your plugins for your new
+  installation using the `kibana-plugin` script. For more information, 
+  see {kibana-ref}/kibana-plugins.html[{kib} plugins].
+
+. Start {kib}.
+
+IMPORTANT: {kib} has a new logging system in 8.0 and the {kib-ref}/upgrade#logging-configuration-changes[log formats have changed]. 
+For additional information, see {kibana-ref}/configuring-logging.html[configuring logging].
```
Comment (debadair) [inline docs/en/install-upgrade/upgrading-kibana.asciidoc:50]:
```suggestion
For additional information, see {kibana-ref}/upgrade.html#configuring-logging[configuring logging].
```

File: docs/en/install-upgrade/upgrading-elasticsearch.asciidoc [hunk 15]
```diff
@@ -0,0 +1,361 @@
...
+
+
+[discrete]
+[[archived-settings]]
+=== Archived settings
+
+If you upgrade an {es} cluster that uses deprecated cluster or index settings
+that are not used in the target version, they are archived and ignored.
+For more information, see 
+{ref}/setup-upgrade.html#archived-settings[Archived settings].
```
Comment (debadair) [inline docs/en/install-upgrade/upgrading-elasticsearch.asciidoc:361]:
```suggestion
{ref}/setup-upgrade.html[Archived settings]. 
```

File: docs/en/install-upgrade/upgrading-stack-on-prem.asciidoc [hunk 16]
```diff
@@ -1,99 +1,19 @@
...
-and reindex your old indices or bring up a new {version} cluster and
-{ref}/reindex-upgrade-remote.html[reindex from remote].
-
-The recommended path is to upgrade to 6.8 before upgrading to {version}. This
-makes it easier to identify the changes you need to make to upgrade and enables
-you to perform a rolling upgrade with no downtime.
\ No newline at end of file
+.. Logstash: upgrade instructions
+.. Beats: upgrade instructions
+.. {agent}: upgrade instructions
```
Comment (debadair) [inline docs/en/install-upgrade/upgrading-stack-on-prem.asciidoc:19]:
```suggestion
.. Beats: {beats-ref}/upgrading.html[upgrade instructions]
.. {agent}: {fleet-guide}/upgrade-elastic-agent.html[upgrade instructions]
```

.. Logstash: {logstash-ref}/upgrading-logstash.html[upgrade instructions]
.. Beats: {beats-ref}/upgrading.html[upgrade instructions]
.. APM: {apm-guide-ref}/upgrade.html[upgrade instructions]
.. {agent}: {fleet-guide}/upgrade-elastic-agent.html[upgrade instructions]

File: docs/en/install-upgrade/upgrading-stack-on-prem.asciidoc [hunk 17]
```diff
@@ -1,99 +1,19 @@
...
-If you are running a version prior to 6.0,
-https://www.elastic.co/guide/en/elastic-stack/6.8/upgrading-elastic-stack.html[upgrade to 6.8]
-and reindex your old indices or bring up a new {version} cluster and
-{ref}/reindex-upgrade-remote.html[reindex from remote].
-
-The recommended path is to upgrade to 6.8 before upgrading to {version}. This
-makes it easier to identify the changes you need to make to upgrade and enables
-you to perform a rolling upgrade with no downtime.
\ No newline at end of file
+.. Logstash: upgrade instructions
```
Comment (debadair) [inline docs/en/install-upgrade/upgrading-stack-on-prem.asciidoc:17]:
```suggestion
.. Logstash: {logstash-ref}/upgrading-logstash.html[upgrade instructions]
```

File: docs/en/install-upgrade/upgrading-kibana.asciidoc [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+  Otherwise, the {kib} instance is assigned a new persistent UUID and becomes a new instance in the monitoring data.
+
+.. Install the appropriate versions of all your plugins for your new
+  installation using the `kibana-plugin` script. For more information, 
+  see {kibana-ref}/kibana-plugins.html[{kib} plugins].
+
+. Start {kib}.
+
+IMPORTANT: {kib} has a new logging system in 8.0 and the {kib-ref}/upgrade#logging-configuration-changes[log formats have changed]. 
+For additional information, see {kibana-ref}/upgrade.html#configuring-logging[configuring logging].
```
Comment (debadair) [inline docs/en/install-upgrade/upgrading-kibana.asciidoc:50]:
```suggestion
For additional information, see {kibana-ref}/upgrade.html[configuring logging].
```

**Pull request merged**